core/node/metadata_calculator/src/helpers.rs

```diff
@@ -550,8 +550,43 @@ impl L1BatchWithLogs {
             MerkleTreeMode::Lightweight => HashSet::new(),
         };
 
+        let load_tree_writes_latency = METRICS.start_load_stage(LoadChangesStage::LoadTreeWrites);
+        let tree_writes = storage
+            .blocks_dal()
+            .get_tree_writes(l1_batch_number)
+            .await?;
+        load_tree_writes_latency.observe();
+
+        let storage_logs = if let Some(tree_writes) = tree_writes {
+            // If tree writes are present in DB then simply use them.
+            let writes = tree_writes.into_iter().map(|tree_write| {
+                let storage_key =
+                    StorageKey::new(AccountTreeId::new(tree_write.address), tree_write.key);
+                TreeInstruction::write(storage_key, tree_write.leaf_index, tree_write.value)
+            });
+            let reads = protective_reads.into_iter().map(TreeInstruction::Read);
+            writes.chain(reads).collect::<Vec<_>>()
```

Comment:
Are tree instructions properly sorted in this case? AFAIU, they should be sorted by `StorageKey`, which doesn't look to be true if `protective_reads` aren't empty.

Reply:
They are not sorted indeed, nice catch!